lib/launchers/dns_utils.rb

```diff
@@ -0,0 +1,71 @@
+#Define two functions to creat/delete dns records, used for osp and vsphere platform
+def create_dns_records(dns_solution, instances_name_prefix, ip_api, ip_api_int, ip_apps)
+  base_domain = nil
+  logger.info("DNS_SOLUTION: #{dns_solution}, PREFIX: #{instances_name_prefix}, IPI_API: #{ip_api}, IP_API_INT: #{ip_api_int}, IP_APPS: #{ip_apps}")
+  case dns_solution
+  when "dyndns"
+    begin
+      tmp_dyn = get_dyn
+      zone = dns_component
+      base_domain = "#{zone}.#{conf[:services, :dyndns, :zone]}"
+      tmp_dyn.dyn_create_a_records("api.#{instances_name_prefix}.#{zone}" , ip_api)
+      unless ip_api_int.nil?
+        tmp_dyn.dyn_create_a_records("api-int.#{instances_name_prefix}.#{zone}" ,ip_api_int)
+      end
+      tmp_dyn.dyn_create_a_records("*.apps.#{instances_name_prefix}.#{zone}", ip_apps)
+      tmp_dyn.publish
+    rescue => e
+      logger.error "Error during creating Dynect DNS records: #{exception_to_string(e)}"
+      raise e
+    end
+  when /^route53/
+    begin
+      logger.info "Creating DNS records in AWS Route 53..."
+      aws_iaas = iaas_by_service(dns_solution.sub(/^route53_?/,"").tap {|svc| svc.empty? && svc << "AWS-CI" } )
```

Comment:
If we create resources on aws, what provide the credential to access aws?

Reply:
the creds are used by the `cucushift-private` repo's config.yaml

Reply:
Why not directly use `AWS-CI` instead of using `dns_solution` for some substitution, or allow users to use which creds as parameters. `dns_solution` should not include any account info.

Reply:
yeah, we can do that...if it's empty it will use `AWS-CI` by default anyway.